mlir/cmake/modules/IRDLToCpp.cmake

```diff
@@ -0,0 +1,9 @@
+function(add_irdl_to_cpp_target target irdl_file)
+  add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${irdl_file}.cpp.inc
+    COMMAND ${MLIR_IRDL_TO_CPP_EXE} ${CMAKE_CURRENT_SOURCE_DIR}/${irdl_file} -o ${CMAKE_CURRENT_BINARY_DIR}/${irdl_file}.cpp.inc
+    DEPENDS ${MLIR_IRDL_TO_CPP_TARGET} ${irdl_file}
+    COMMENT "Building ${irdl_file}..."
+  )
+  add_custom_target(${target} ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${irdl_file}.cpp.inc)
```

Comment:
Do we really need both `add_custom_command` and `add_custom_target`? The latter has a `COMMAND` argument one can use directly.

I also see all tablegen'ed files explicitly marked as [`GENERATED'](https://cmake.org/cmake/help/v3.31/prop_sf/GENERATED.html#prop_sf:GENERATED) https://github.com/llvm/llvm-project/blob/84cb08e11866dda721e9a1b5a39c4e963fc36c32/mlir/cmake/modules/AddMLIR.cmake#L124-L125

Reply:
The separate custom command is useful because target does not track file output: if there is no separate command, the output files are rebuilt every time (due to a dependency CMake infers I presume?).

Adding GENERATED seems unnecessary, as CMake adds it to the output of commands anyway? But I found that depending on the executable directly instead of the target fixes the problem, albeit in a slightly sledgehammer way. If that sounds good to you, we can move on with merging.